crowdfunding/projects/views.py

```diff
@@ -1,3 +1,87 @@
 from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from django.http import Http404
+from rest_framework import status, generics, permissions, filters
+from django_filters.rest_framework import DjangoFilterBackend
+
+from .models import Project, Pledge
+from .serializers import ProjectSerializer, PledgeSerializer, ProjectDetailSerializer
+from .permissions import IsOwnerOrReadOnly, IsSupporterOrReadOnly
+
 
 # Create your views here.
+# class ProjectList(APIView):
+    # permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+    # 
+    # def get(self, request):
+        # projects = Project.objects.all()
+        # serializer = ProjectSerializer(projects, many=True) #given many of the lists
+        # return Response(serializer.data)
+    # 
+class ProjectList(generics.ListCreateAPIView):
+
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
+    filter_backends = [DjangoFilterBackend, filters.SearchFilter,]
+    filterset_fields = ["owner", "is_open"]
+    search_fields = ["title", "description"]
+  
+    
+    def post(self, request):
+        serializer = ProjectSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save(owner=request.user)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
not sure you needed this post method, because it should be handled by the generic that you are using.